koku/masu/database/ocp_report_db_accessor.py

```diff
@@ -1002,7 +1002,21 @@ def populate_cluster_table(self, provider, cluster_id, cluster_alias):
         """Get or create an entry in the OCP cluster table."""
         with schema_context(self.schema):
             LOG.info(log_json(msg="fetching entry in reporting_ocp_cluster", provider_uuid=provider.uuid))
-            cluster = OCPCluster.objects.filter(provider_id=provider.uuid).first()
+            clusters = OCPCluster.objects.filter(provider_id=provider.uuid)
+            if len(clusters) > 1:
+                for cluster_iter in clusters:
+                    if cluster_iter.cluster_alias != provider.name:
+                        LOG.info(
+                            log_json(
+                                msg="attempting to delete duplicate entry in reporting_ocp_cluster",
+                                provider_uuid=provider.uuid,
+                            )
+                        )
+                        cluster_iter.delete()
+                    else:
+                        cluster = cluster_iter
+            else:
+                cluster = clusters.first()
```

Comment:
Of course, there are always many ways to do something, but we can simplify this a bit by utilizing the ORM:
```
            if clusters.count() > 1:
                clusters_to_delete = clusters.exclude(cluster_alias=provider.name)
                LOG.info(
                    log_json(
                        msg="attempting to delete duplicate entry in reporting_ocp_cluster",
                        provider_uuid=provider.uuid,
                    )
                )
                clusters_to_delete.delete()

            cluster = clusters.first()
```

This way, there is less branching, and more of making the db do all the work here.